crates/cheatnet/src/runtime_extensions/forge_runtime_extension/mod.rs

```diff
@@ -750,47 +750,47 @@ fn concat_u128_bytes(low: &[u8; 32], high: &[u8; 32]) -> [u8; 32] {
 }
 
 #[must_use]
-pub fn get_all_execution_resources(runtime: ForgeRuntime) -> UsedResources {
-    let runtime_execution_resources = runtime
+pub fn update_top_call_execution_resources(runtime: ForgeRuntime) -> UsedResources {
+    let starknet_runtime = runtime
         .extended_runtime
         .extended_runtime
         .extended_runtime
-        .extended_runtime
-        .hint_handler
-        .resources
-        .clone();
-    let runtime_l1_to_l2_messages = runtime
-        .extended_runtime
-        .extended_runtime
-        .extended_runtime
-        .extended_runtime
-        .hint_handler
-        .l2_to_l1_messages;
+        .extended_runtime;
+    let test_code_execution_resources = starknet_runtime.hint_handler.resources.clone();
+    let test_code_l1_to_l2_messages = starknet_runtime.hint_handler.l2_to_l1_messages;
 
     let runtime_call_info = CallInfo {
         execution: CallExecution {
-            l2_to_l1_messages: runtime_l1_to_l2_messages,
+            l2_to_l1_messages: test_code_l1_to_l2_messages,
             ..Default::default()
         },
+        inner_calls: starknet_runtime.hint_handler.inner_calls,
```

Comment:
It fixes calculating l1->l2 messages in calls directly from tests that do not use the `StarknetRuntime` used by the test code (== are not caught by `CallToBlockifierRuntime`) e.g. in `deploy_syscall`